File: src/java/foundgine-core/src/main/java/com/foundgine/core/semantic/authorization/SemanticOperationGraphAuthorizationResult.java
package com.foundgine.core.semantic.authorization;
import com.foundgine.core.semantic.*; import com.foundgine.core.semantic.ir.graph.*; import java.util.*;
public record SemanticOperationGraphAuthorizationResult(SemanticOperationGraph graph,SemanticAuthorizationEvidence evidence){public SemanticOperationGraphAuthorizationResult{Objects.requireNonNull(graph);Objects.requireNonNull(evidence);}public void ensureMatches(SemanticContractSnapshot c){evidence.ensureMatches(c);}}

File: src/csharp/Foundgine.Core/Semantic/Authorization/SemanticOperationGraphAuthorizationResult.cs
using Foundgine.Core.Semantic.IR;
using Foundgine.Core.Semantic.IR.Graph;

namespace Foundgine.Core.Semantic.Authorization;

/// <summary>
/// Authorization output for a canonical semantic operation graph. The graph and
/// evidence are produced together so a caller cannot accidentally detach the
/// authorized graph from the contract against which it was evaluated.
/// </summary>
public sealed record SemanticOperationGraphAuthorizationResult(
    SemanticOperationGraph Graph,
    SemanticAuthorizationEvidence Evidence)
{
    public void EnsureMatches(SemanticContractSnapshot contract)
    {
        ArgumentNullException.ThrowIfNull(contract);
        Evidence.EnsureMatches(contract);
    }
}
